graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -44,6 +51,138 @@ def _are_filter_fields_uniquely_indexed(filter_fields, unique_indexes):
+                             .format(filter_operator))
+
+    # If the interval of values passing through the filters is empty, no results will be
+    # returned. This happens if the interval's upper bound is smaller than the lower bound.
+    if query_interval[0] > query_interval[1]:
+        field_selectivity = Selectivity(kind=ABSOLUTE_SELECTIVITY, value=0.0)
+        return field_selectivity
+
+    if not domain_interval[0] <= query_interval[0] <= query_interval[1] <= domain_interval[1]:
+        raise AssertionError(u'Query interval {} is not'
+                             u'a subset of the given domain interval {}.'
+                             .format(query_interval, domain_interval))
+
+    # Assumption: the values of the integers being filtered are evenly distributed among the domain
+    # of valid values.
+    query_interval_size = query_interval[1] - query_interval[0] + 1
+    domain_interval_size = domain_interval[1] - domain_interval[0] + 1
+    fraction_of_domain_queried = query_interval_size / domain_interval_size
+
+    field_selectivity = Selectivity(
+        kind=FRACTIONAL_SELECTIVITY,
+        value=query_interval_size / domain_interval_size
+    )
+    return field_selectivity
+
+
+def _estimate_inequality_filter_selectivity(
+    schema_graph, statistics, filter_info, parameters, location_name
+):
+    """Calculate the selectivity of a specific inequality filter at a given location.
+
+    Args:
+        schema_graph: SchemaGraph object
+        statistics: Statistics object
+        filter_info: FilterInfo object, filter on the location being filtered
+        parameters: dict, parameters with which query will be executed
+        location_name: string, type of the location being filtered
+
+    Returns:
+        Selectivity object, the selectivity of a specific inequality filter at a given location.
+    """
+    filter_operator = filter_info.op_name
+    if filter_operator not in INEQUALITY_OPERATORS:
+        raise ValueError(u'Inequality filter selectivity estimator received a filter'
+                         u'with non-inequality filter operator {}: {} {}'
+                         .format(filter_operator, filter_info, location_name))
+
+    all_selectivities = []
+    for field_name in filter_info.fields:
+        field_selectivity = Selectivity(kind=FRACTIONAL_SELECTIVITY, value=1.0)
+
+        # TODO(vlad): Improve inequality estimation by implementing histograms.
+        if field_name == 'uuid':
+            # Instead of working with UUIDs, we convert each occurence of UUID to its corresponding
+            # integer representation.
+            uuid_domain = (MIN_UUID_INT, MAX_UUID_INT)
+
+            parameter_values = tuple(
+                _try_obtain_parameter_value(filter_argument, parameters)
+                for filter_argument in filter_info.args
+            )
+            parameter_values_as_integers = tuple(
```

Comment:
could we convert these two tuples into a single tuple? e.g. chaining the function calls.

Reply:
It starts looking like badly-written Lisp really quick though: ` _convert_uuid_string_to_int(parameters[get_parameter_name(filter_argument)])`.